app/scripts/lib/multichain-api/caip25permissions.test.ts

```diff
@@ -215,5 +234,397 @@ describe('endowment:caip25', () => {
     });
   });
 
-  // it.todo('permission validator');
+  describe('permission validator', () => {
+    const findNetworkClientIdByChainId = jest.fn();
+    const { validator } = caip25EndowmentBuilder.specificationBuilder({
+      findNetworkClientIdByChainId,
+    });
+
+    it('throws an error if there is not exactly one caveat', () => {
+      expect(() => {
+        validator({
+          caveats: [
+            {
+              type: 'caveatType',
+              value: {},
+            },
+            {
+              type: 'caveatType',
+              value: {},
+            },
+          ],
+          date: 1234,
+          id: '1',
+          invoker: 'test.com',
+          parentCapability: Caip25EndowmentPermissionName,
+        });
+      }).toThrow(new Error('missing required caveat'));
```

Comment:
Is there a todo to improve this error?

Reply:
i think we're going to have to just comb through all the changes in the feature branch to make sure we catch all these because not all of them have TODO comments